Brokerages/Binance/BinanceBrokerage.Messaging.cs

```diff
@@ -0,0 +1,317 @@
+            while (_messageBuffer.Any())
+            {
+                WebSocketMessage e;
+                _messageBuffer.TryDequeue(out e);
+                OnMessageImpl(this, e);
+                OnTickerMessageImpl(this, e);
+            }
+            Log.Trace("BinanceBrokerage.Messaging.UnlockStream(): Stream Unlocked.");
+            // Once dequeued in order; unlock stream.
+            _streamLocked = false;
+        }
+
+        private void OnMessageImpl(object sender, WebSocketMessage e)
+        {
+            try
+            {
+                var wrapped = JObject.Parse(e.Message);
+                var message = wrapped.GetValue("data").ToObject<Messages.BaseMessage>();
+                switch (message.Event)
+                {
+                    case "executionReport":
+                        var upd = wrapped.GetValue("data").ToObject<Messages.Execution>();
+                        if (upd.ExecutionType.Equals("TRADE", StringComparison.OrdinalIgnoreCase))
+                        {
+                            OnFillOrder(upd);
+                        }
+                        break;
+                    default:
+                        return;
+                }
+            }
+            catch (Exception exception)
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, -1, $"Parsing wss message failed. Data: {e.Message} Exception: {exception}"));
+                throw;
+            }
+        }
+
+        private void OnTickerMessageImpl(object sender, WebSocketMessage e)
+        {
+            try
+            {
+                var wrapped = JObject.Parse(e.Message);
+                var message = wrapped.GetValue("data").ToObject<Messages.BaseMessage>();
+                switch (message.Event)
+                {
+                    case "depthUpdate":
+                        var updates = wrapped.GetValue("data").ToObject<Messages.OrderBookUpdateMessage>();
+                        OnOrderBookUpdate(updates);
+                        break;
+                    case "trade":
+                        var trade = wrapped.GetValue("data").ToObject<Messages.Trade>();
+                        EmitTradeTick(
+                            _symbolMapper.GetLeanSymbol(trade.Symbol),
+                            Time.UnixMillisecondTimeStampToDateTime(trade.Time),
+                            trade.Price,
+                            trade.Quantity
+                        );
+                        break;
+                    default:
+                        return;
+                }
+            }
+            catch (Exception exception)
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, -1, $"Parsing wss message failed. Data: {e.Message} Exception: {exception}"));
+                throw;
+            }
+        }
+
+        private void OnOrderBookUpdate(OrderBookUpdateMessage ticker)
```

Comment:
Perhaps outside the scope of this PR, but it would be nice to have an `OrderBookCollection` or `OrderBookManager` of sorts as a self-contained component. Often times, such is the case with binance,  you can bind to their L2 data feed _without_ credentials. This could allow an algorithm trading on, for example, coinbase, to still get the binance quote books for usage within the algorithm.